src/Model/Documentable/DocumentableProductTrait.php

```diff
@@ -106,23 +91,30 @@ protected function addChannelsInDocument(ResultInterface $document): ResultInter
         return $document;
     }
 
-    /**
-     * @param ResultInterface $document
-     *
-     * @return ResultInterface
-     */
     protected function addPricesInDocument(ResultInterface $document): ResultInterface
     {
         /** @var Channel $channel */
         foreach ($this->getChannels() as $channel) {
-            /** @var ProductVariant $variant */
-            if ($variant = $this->getCheapestVariantForChannel($channel)) {
-                $price = $variant->getChannelPricingForChannel($channel);
-
+            $cheapestVariantChannelPricing = $this->getCheapestVariantChannelPricingForChannel($channel);
+            if ($cheapestVariantChannelPricing !== null) {
                 /** @var CurrencyInterface $currency */
                 foreach ($channel->getCurrencies() as $currency) {
-                    $document->addPrice($channel->getCode(), $currency->getCode(), $price->getPrice());
-                    if ($originalPrice = $price->getOriginalPrice()) {
+                    $variantPrice = $cheapestVariantChannelPricing->getPrice();
+                    if ($variantPrice < $cheapestVariantChannelPricing->getMinimumPrice()) {
+                        $variantPrice = $cheapestVariantChannelPricing->getMinimumPrice();
+                    }
```

Comment:
```suggestion
                    $variantPrice = min($cheapestVariantChannelPricing->getPrice(), $cheapestVariantChannelPricing->getMinimumPrice());
```
As a side note, isn't the condition in the if() of the original code backward? As it was, it looks like it got the HIGHEST price, not the lowest.

Reply:
> As a side note, isn't the condition in the if() of the original code backward? As it was, it looks like it got the HIGHEST price, not the lowest.

Why?

Reply:
We don't have to check for the min price between the price and minimumPrice, we have to take the minimum price only if the price is less than this.